docs/ossm/ossm2-vs-ossm3.md

```diff
@@ -0,0 +1,103 @@
+
+## `Istio` replaces `ServiceMeshControlPlane`
+
+While OpenShift Service Mesh 2 used a resource called `ServiceMeshControlPlane` to configure Istio, OpenShift Service Mesh 3 uses a resource called `Istio`. 
+
+The `Istio` resource contains a `spec.values` field that derives its schema from Istio’s Helm chart values. While this is a different configuration schema than `ServiceMeshControlPlane` uses, the fact that it is derived from Istio’s configuration means that configuration examples from the community Istio documentation can often be applied directly to Red Hat OpenShift Service Mesh’s `Istio` resource. The `spec.values` field in the `IstioOperator` resource (which is not part of OpenShift Service Mesh) has a similar format. The `Istio` resource provides an additional validation schema enabling the ability to explore the resource using the OpenShift CLI command `oc explain istios.spec.values`.
+
+## New resource: `IstioCNI`
+
+The Istio CNI node agent is used to configure traffic redirection for pods in the mesh. It runs as a DaemonSet, on every node, with elevated privileges. The Istio CNI agent has a lifecycle that is independent of Istio control planes, and must be upgraded separately. 
+
+In OpenShift Service Mesh 2, service meshes were namespace scoped by default and thus each mesh included its own instance of Istio CNI. As all OpenShift Service Mesh 3 meshes are cluster-wide, there can only be one version of Istio CNI per cluster, regardless of how many service mesh control planes are on the cluster.
+
+For these reasons, the OpenShift Service Mesh 3 operator manages Istio CNI with a separate resource called `IstioCNI`. A single instance of this resource is shared by all Istio control planes (managed by `Istio` resources). The `IstioCNI` resource must be upgraded before individual control planes (`Istio` resources) are upgraded.
+
+## Scoping of the Mesh: Discovery Selectors and labels replace `ServiceMeshMemberRoll` and `ServiceMeshMember`
+
+OpenShift Service Mesh 2 used the two resources `ServiceMeshMemberRoll` and `ServiceMeshMember` to indicate which namespaces were to be included in the mesh. When a mesh was created, it would only be scoped to the namespaces listed in the `ServiceMeshMemberRoll` or containing a `ServiceMeshMember` instance. This made it simple to include multiple service meshes in a cluster with each mesh tightly scoped, referred to as a “multitenant” configuration. 
+
+In OpenShift Service Mesh 2.4, a “cluster-wide” mode was introduced to allow a mesh to be cluster-scoped, with the option to limit the mesh using an Istio feature called `discoverySelectors`, which limits the scope of the mesh to a set of namespaces defined with a [label selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/). This is aligned with how community Istio worked, and allowed Istio to manage cluster-level resources.
```

Comment:
> which limits the scope of the mesh ...

This is not perfectly exact. Yes, in 2.4 we could say that the SMMR/SMM limited the scope of the mesh, but in 3.x (and 2.x cluster-wide mode), there is no single concept of "scope". Instead, scope is split into discovery and injection. In 2.x multitenant mode the SMMR affected both at the same time. In 2.x cluster-wide and 3.x, you limit discovery via `discoverySelectors` and injection via namespace/workload labels.

Reply:
"scope of the mesh" is probably an over simplification, as you're right that injection matters too (the [injection doc PR](https://github.com/openshift-service-mesh/sail-operator/pull/152) is relevant here too). I can change "limits the scope" to "limits the Istio control plane's visibility", which is more precise.

In 2.x though, at least as [we documented](https://docs.openshift.com/container-platform/4.17/service_mesh/v2x/prepare-to-deploy-applications-ossm.html#ossm-automatic-sidecar-injection_deploying-applications-ossm), you still needed to add an annotation or label to workloads for injection to happen. So, Discovery Selectors doesn't sound too different in practice (even though the Istiods are scoped differently)? Perhaps it was more different under the hood...